BAT/BCAux.h

```diff
@@ -119,6 +119,47 @@ void DrawKnowledgeUpdate(BCHistogramBase& prior, BCHistogramBase& posterior, boo
  * @return Number of plots printed */
 unsigned PrintPlots(std::vector<BCH1D>& h1, std::vector<BCH2D>& h2, std::string filename, unsigned hdiv = 1, unsigned vdiv = 1);
 
+/**
+ * A trash to keep heap-allocated objects of type T alive until the
+ * trash goes out of scope. Then they are deleted. Ownership is not
+ * transferred in copies, only during a swap.
+ */
+template <typename T>
+class BCTrash
```

Comment:
BCTrash might not be an optimal name for this class:

    BCTrash.Put(...)

sounds like "delete this", instead of "keep this alive". :-)